### PR TITLE
Booking endpoint fix

### PR DESCRIPTION
**Pull request by zsinx6**
#### What
Fix booking creation endpoint and models.

#### Why
Make sure the booking creation endpoint accounts for all fields and improve testing.

`Task`: https://omnivector-solutions.monday.com/boards/1204470680/pulses/1579574012

### CODE REVIEW

File: backend/tests/api/test_booking.py [hunk 1]
```diff
@@ -101,17 +131,86 @@ async def test_bookings_all(
...
     ]
+
+
+@mark.asyncio
+@mock.patch("lm_backend.api.booking.edit_counts")
+@mock.patch("lm_backend.api.booking.map_bookings")
+@database.transaction(force_rollback=True)
+async def test_booking_create(
+    edit_counts_mock, map_bookings_mock, backend_client, some_config_rows, insert_objects
+):
```
Comment (jamesbeedy) [inline backend/tests/api/test_booking.py:162]:
```suggestion
):
    """This test proves that a booking can be created by showing that edit_counts and map_bookings
    are awaited one time and the response status is 200.
    """
```

File: backend/tests/api/test_booking.py [hunk 2]
```diff
@@ -101,17 +131,86 @@ async def test_bookings_all(
...
+    map_bookings_mock.assert_awaited_once()
+
+
+@mark.asyncio
+@mock.patch("lm_backend.api.booking.edit_counts")
+@mock.patch("lm_backend.api.booking.map_bookings")
+@database.transaction(force_rollback=True)
+async def test_booking_create_negative_booked_error(
+    edit_counts_mock, map_bookings_mock, backend_client, some_config_rows, some_licenses, insert_objects
+):
```
Comment (jamesbeedy) [inline backend/tests/api/test_booking.py:182]:
```suggestion
):
    """This test proves that a 400 response code is returned when a `-` (negative) booking creation is attempted
    by checking the response code of a booking request containing a negative booking and by asserting that
    edit_counts() and map_bookings() were not awaited.
    """
```

File: backend/tests/api/test_booking.py [hunk 3]
```diff
@@ -101,17 +131,86 @@ async def test_bookings_all(
...
+@database.transaction(force_rollback=True)
+async def test_booking_delete(
+    backend_client, some_config_rows, some_licenses, some_booking_rows, insert_objects
+):
+    await insert_objects(some_licenses, table_schemas.license_table)
+    await insert_objects(some_config_rows, table_schemas.config_table)
+    await insert_objects(some_booking_rows, table_schemas.booking_table)
+
+    resp = await backend_client.delete("/api/v1/booking/book/helloworld")
+    assert resp.status_code == status.HTTP_200_OK
```
Comment (jamesbeedy) [inline backend/tests/api/test_booking.py]:
```suggestion
@mark.asyncio
@database.transaction(force_rollback=True)
async def test_booking_delete(
    backend_client, some_config_rows, some_licenses, some_booking_rows, insert_objects,
):
    """This test proves that the correct response is returned (200) when a booking
    is successfully deleted and that the functions map_bookings() and edit_counts()
    are awaited.
    """
    await insert_objects(some_licenses, table_schemas.license_table)
    await insert_objects(some_config_rows, table_schemas.config_table)
    await insert_objects(some_booking_rows, table_schemas.booking_table)

    resp = await backend_client.delete("/api/v1/booking/book/helloworld")
    assert resp.status_code == status.HTTP_200_OK
```

Comment (zsinx6) [reply backend/tests/api/test_booking.py]: edit_counts and map_bookings are not called in the delete endpoint, then they are not needed to be mocked

File: backend/tests/api/test_booking.py [hunk 4]
```diff
@@ -101,17 +131,86 @@ async def test_bookings_all(
...
+    backend_client, some_config_rows, some_licenses, insert_objects
+):
+    await insert_objects(some_licenses, table_schemas.license_table)
+    await insert_objects(some_config_rows, table_schemas.config_table)
+    features = BookingFeature(booked=1000, product_feature="hello.world")
+    booking = Booking(job_id=1, features=[features], lead_host="host1", user_name="user1")
+    resp = await backend_client.put("/api/v1/booking/book", json=booking.dict())
+
+    assert resp.status_code == status.HTTP_400_BAD_REQUEST
+    assert "<= total" in resp.json()["detail"]
```
Comment (jamesbeedy) [inline backend/tests/api/test_booking.py]:
```suggestion
@mark.asyncio
@database.transaction(force_rollback=True)
async def test_booking_create_booked_greater_than_total(
        backend_client, some_config_rows, some_licenses, insert_objects,
):
    """This test proves that the correct response (400) is returned when a booking
    request exceeds the total available by asserting that the response detail contains
    the string "<= total" and asserting that the functions map_booking() and edit_counts()
    are not awaited.
    """
    await insert_objects(some_licenses, table_schemas.license_table)
    await insert_objects(some_config_rows, table_schemas.config_table)
    features = BookingFeature(booked=1000, product_feature="hello.world")
    booking = Booking(job_id=1, features=[features], lead_host="host1", user_name="user1")
    resp = await backend_client.put("/api/v1/booking/book", json=booking.dict())

    assert resp.status_code == status.HTTP_400_BAD_REQUEST
    assert "<= total" in resp.json()["detail"]
```

**Review by jamesbeedy — CHANGES_REQUESTED** [review]
Left some inline comments

File: backend/tests/api/test_booking.py (hunk 4 above)
Comment (zsinx6) [reply backend/tests/api/test_booking.py]: This one is better to just call the functions, instead of using mock, because the 400 will raise from then

**Comment by zsinx6** [comment]
> Left some inline comments

I've fixed some suggestions to get the docstrings, but the mocks are not needed there

**Review by jamesbeedy — APPROVED** [review]
👍

**Pull request merged**